app/src/main/java/com/github/se/polyfit/viewmodel/post/CreatePostViewModel.kt

```diff
@@ -0,0 +1,71 @@
+package com.github.se.polyfit.viewmodel.post
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import com.github.se.polyfit.data.remote.firebase.PostFirebaseRepository
+import com.github.se.polyfit.data.repository.MealRepository
+import com.github.se.polyfit.model.meal.Meal
+import com.github.se.polyfit.model.post.Location
+import com.github.se.polyfit.model.post.Post
+import com.github.se.polyfit.model.post.UnmodifiablePost
+import dagger.hilt.android.lifecycle.HiltViewModel
+import java.time.LocalDate
+import javax.inject.Inject
+
+@HiltViewModel
+class CreatePostViewModel
+@Inject
+constructor(
+    private val mealRepository: MealRepository,
+    private val postFirebaseRepository: PostFirebaseRepository,
+) : ViewModel() {
+  private val _post = Post.default()
+
+  val post: UnmodifiablePost
+    get() = _post
+
+  suspend fun getRecentMeals() = mealRepository.getAllMeals().sortedBy { it.createdAt }
```

Comment:
nit: Getting all the meals from a user could potentially be expensive and only getting meals within a certain amount of time could be better. That being said, for the scope of our project, it's probably fine.

Reply:
Discussed offline: to create a new Post, only the last five recent meals will be shown as potential options, instead of all of them.